### PR TITLE
[AMDGPU][GISel] Use datalayout alignment for buffer-load legalization

### PR DESCRIPTION
**Pull request by cdevadas**
Making the legalization of buffer loads similar to the SelectionDAG.

### CODE REVIEW

File: llvm/lib/CodeGen/GlobalISel/Utils.cpp [hunk 1]
```diff
@@ -1906,3 +1906,10 @@ bool llvm::isGuaranteedNotToBeUndef(Register Reg,
   return ::isGuaranteedNotToBeUndefOrPoison(Reg, MRI, Depth,
                                             UndefPoisonKind::UndefOnly);
 }
+
+Type *llvm::getTypeForLLT(LLT Ty, LLVMContext &C) {
+  if (Ty.isVector())
+    return FixedVectorType::get(IntegerType::get(C, Ty.getScalarSizeInBits()),
+                                Ty.getNumElements());
```
Comment (arsenm) [inline llvm/lib/CodeGen/GlobalISel/Utils.cpp:1913]: Use VectorType and getElementCount

**Review by arsenm — APPROVED** [review]
Still not sure if we should just use 4, but this matches the DAG behavior.

It would make more sense to move the s buffer intrinsics to use pointers and take the align attribute from the parameter

File: llvm/test/CodeGen/AMDGPU/GlobalISel/legalize-llvm.amdgcn.s.buffer.load.mir [hunk 2]
```diff
@@ -272,7 +272,7 @@ body:             |
     ; GFX67-NEXT: {{  $}}
     ; GFX67-NEXT: [[COPY:%[0-9]+]]:_(<4 x s32>) = COPY $sgpr0_sgpr1_sgpr2_sgpr3
     ; GFX67-NEXT: [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 0
-    ; GFX67-NEXT: [[AMDGPU_S_BUFFER_LOAD:%[0-9]+]]:_(<4 x s32>) = G_AMDGPU_S_BUFFER_LOAD [[COPY]](<4 x s32>), [[C]](s32), 0 :: (dereferenceable invariant load (s96), align 4)
+    ; GFX67-NEXT: [[AMDGPU_S_BUFFER_LOAD:%[0-9]+]]:_(<4 x s32>) = G_AMDGPU_S_BUFFER_LOAD [[COPY]](<4 x s32>), [[C]](s32), 0 :: (dereferenceable invariant load (s96), align 16)
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/legalize-llvm.amdgcn.s.buffer.load.mir:275]: We might be setting the wrong alignment field, I thought the MMO didn't print the align when it was the same as the size

Comment (cdevadas) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/legalize-llvm.amdgcn.s.buffer.load.mir:275]: I noticed that too. I'll take a look and can post a separate PR if any changes needed.

**Pull request merged**